apps/financial-aid/web-veita/src/components/StateModal/StateModal.tsx

```diff
@@ -52,9 +46,17 @@ const StateModal = ({
     amount?: number,
     rejection?: string,
   ) => {
-    changeApplicationState(application, state, amount, rejection)
+    const updatedApplication = await changeApplicationState(
+      application,
+      state,
+      amount,
+      rejection,
+    )
+    if (updatedApplication) {
+      setApplication(updatedApplication)
+    }
+
```

Comment:
Are we doing this in many places? Just wondering if it makes sense to extract this or not